source/Irrlicht/COSOperator.cpp

```diff
@@ -107,19 +107,24 @@ void COSOperator::copyToClipboard(const c8 *text) const
 const c8* COSOperator::getTextFromClipboard() const
 {
 #if defined(_IRR_WINDOWS_API_)
+	if (!IsClipboardFormatAvailable(CF_TEXT))
+		return 0;
 	if (!OpenClipboard(NULL))
 		return 0;
 
-	wchar_t * buffer = 0;
-
-	HANDLE hData = GetClipboardData( CF_UNICODETEXT );
-	buffer = (wchar_t*) GlobalLock( hData );
-
-	core::wStringToMultibyte(ClipboardBuf, buffer);
+	char * buffer = 0;
 
-	GlobalUnlock( hData );
+	HANDLE hData = GetClipboardData( CF_TEXT );

```

Comment:
[Microsoft documentation](https://learn.microsoft.com/en-us/windows/win32/dataxchg/standard-clipboard-formats) says this format is ANSI text, which means whatever random codepage the user has and _not_ necessarily Unicode.

what was wrong with using `CF_UNICODETEXT`?

Reply:
Only first unicode char be pasted if using CF_UNICODETEXT.

Reply:
That sounds like a conversion issue, I'll try to investigate the bug.

Reply:
FWIW, Wine does some polling to open the clipboard. This might be related to pending locks from other threads.

https://github.com/wine-mirror/wine/blob/master/dlls/user32/tests/clipboard.c#L38-L65